### PR TITLE
Api 37585 remove firstname requirement

### PR DESCRIPTION
**Pull request by mchristiansonVA**
## Summary

- *This work is behind a feature toggle (flipper): NO*
- Removes check for missing first_name from target_veteran.
- Adds logic to base class to check for missing name scenarios.
- Adds checks to conditionally require first OR last name to the following endpoints:
/veterans/{veteranId}/claims/{id}/5103 
/veterans/{veteranId}/526/generatePDF/minimum-validations
/veterans/{veteranId}/2122
/veterans/{veteranId}/2122/validate
/veterans/{veteranId}/2122a
/veterans/{veteranId}/2122a/validate
- Adds checks to conditionally require first AND last name to the following endpoints:
/veterans/{veteranId}/526
/veterans/{veteranId}/526/synchronous
/veterans/{veteranId}/526/validate

## Related issue(s)
[API-37585](https://jira.devops.va.gov/browse/API-37585)

## Testing done

- [x] *New code is covered by unit tests*
- Target veteran logic formerly checked for presence of first_name and raised a 422 if not found.
- Updated logic removes first_name check, adds conditional logic to check for the presence of veteran name elements to various endpoints.
- Tested missing first name, last name, or both for endpoints to which check was added.

## What areas of the site does it impact?
/veterans/{veteranId}/claims/{id}/5103 
/veterans/{veteranId}/526/generatePDF/minimum-validations
/veterans/{veteranId}/2122
/veterans/{veteranId}/2122/validate
/veterans/{veteranId}/2122a
/veterans/{veteranId}/2122a/validate
/veterans/{veteranId}/526
/veterans/{veteranId}/526/synchronous
/veterans/{veteranId}/526/validate

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/spec/requests/v2/veterans/power_of_attorney_org_request_spec.rb [hunk 1]
```diff
@@ -71,6 +71,76 @@
...
+                ssn: '796043735',
+                participant_id: '600061742',
+                mpi: OpenStruct.new(
+                  icn: '1012832025V743496',
+                  profile: OpenStruct.new(ssn: '796043735')
+                )
+              )
+            end
+
+            it 'returns a success response' do
```
Comment (tycol7) [inline modules/claims_api/spec/requests/v2/veterans/power_of_attorney_org_request_spec.rb:127]: Do we mean to say 'returns an error response'?

**Review by tycol7 — COMMENTED** [review]
Looking good - functionally works and all AC met. Just some naming issues in the tests and one code suggestion.

File: modules/claims_api/app/controllers/claims_api/v2/veterans/base.rb [hunk 2]
```diff
@@ -22,6 +22,24 @@ def validate_json_schema(form_number = self.class::FORM_NUMBER)
...
+              raise_exception_name_error('Missing first and last name')
+            elsif target_veteran[:first_name].blank?
+              raise_exception_name_error('Missing first name')
+            elsif target_veteran[:last_name].blank?
+              raise_exception_name_error('Missing last name')
+            end
+          elsif target_veteran[:first_name].blank? && target_veteran[:last_name].blank?
+            raise_exception_name_error('Must have either first or last name')
+          end
+        end
```
[the remark this answers is not in the document]
Comment (mchristiansonVA) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/base.rb]: Thanks for the suggestion @tycol7. I modified the method a bit for clarity and to reduce repetition. Let me know if you think it still needs further changes to improve clarity.

[the remark this answers is not in the document]
Comment (tycol7) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/base.rb]: Looks good to me!

**Review by stiehlrod — APPROVED** [review]
Screenshot 2024-07-16 at 8 38 58 AM

**Pull request merged**